drivers/sx126x/sx126x_radio_hal.c

```diff
@@ -0,0 +1,647 @@
+
+        break;
+
+        case IEEE802154_HAL_OP_CCA:
+            DEBUG("[sx126x] netdev: HAL_OP_CCA (CAD Detection state)\n");
+            dev->cad_detected = 0;
+            _set_state(dev, STATE_IDLE);
+            sx126x_set_cad(dev);
+
+        break;
+
+        default:
+        assert(false);
+        break;
+
+    }
+    return 0;
+}
+
+static int _confirm_op(ieee802154_dev_t *hal, ieee802154_hal_op_t op, void *ctx){
+    sx126x_t *dev = hal->priv;
+    (void)op;
+    (void)ctx;
+    ieee802154_tx_info_t *info = ctx;
+    bool eagain = false;
+    sx126x_state_t state;
+    _get_state(dev, &state);
+
+    
+switch (op){
+    case IEEE802154_HAL_OP_TRANSMIT:
+       if (info) {
+            info->status = (dev->cad_detected) ? TX_STATUS_MEDIUM_BUSY : TX_STATUS_SUCCESS;
+        }
+    while(state == STATE_TX)
+        _get_state(dev, &state);
+        
+    break;
+
+    case IEEE802154_HAL_OP_SET_RX:
+        if(state!=STATE_RX) 
+            eagain = true;
+    break;
+
+    case IEEE802154_HAL_OP_SET_IDLE:
+        eagain = (state != STATE_IDLE);
+        break;
+
+    case IEEE802154_HAL_OP_CCA:
+            *((bool*) ctx) = !dev->cad_detected;
+    break;
+
+    default:
+        eagain = false;
+        assert(false);
+        break;
+    
+}
+
+    if (eagain) {
+        DEBUG("EAGAIN");
+        return -EAGAIN;
+    }
+
+    return 0;
+}
+
+static int _len(ieee802154_dev_t *hal){
+    (void)hal;
+    sx126x_t *dev = hal->priv;
+    sx126x_rx_buffer_status_t rx_buffer_status;
+    sx126x_get_rx_buffer_status(dev, &rx_buffer_status);
+    dev->size = rx_buffer_status.pld_len_in_bytes;
+    return dev->size;
+}
+
+static int _read(ieee802154_dev_t *hal, void *buf, size_t max_size, ieee802154_rx_info_t *info)
+{
+    (void)hal;
+    (void)buf;
+    (void)info;
+
+    sx126x_t* dev = hal->priv;
+
+        /* Getting information about last received packet */
+    netdev_lora_rx_info_t *packet_info = (void*)info;
+    sx126x_rx_buffer_status_t rx_buffer_status;
+    sx126x_pkt_status_lora_t pkt_status;
+    sx126x_get_rx_buffer_status(dev, &rx_buffer_status);
+
+    dev->size = rx_buffer_status.pld_len_in_bytes;
+    sx126x_get_lora_pkt_status(dev, &pkt_status);
+    if (packet_info) {
+        packet_info->snr = pkt_status.snr_pkt_in_db;
+        packet_info->rssi = pkt_status.rssi_pkt_in_dbm;
+    }
+
+      /* Put PSDU to the output buffer */
+    
+    if (buf == NULL) {
+        return dev->size;
+    }
+
+    if (dev->size > max_size) {
+        return -ENOBUFS;
+    }
+
+    if (dev->size < 3) {
+        return -EBADMSG;
+    }
+    sx126x_read_buffer(dev, rx_buffer_status.buffer_start_pointer, (uint8_t*)buf, dev->size);
+    DEBUG("[sx126x] first 3 bytes of received packet: %d %d %d\n", *(uint8_t*)buf, *((uint8_t*)buf+1), *((uint8_t*)buf+2));
+    return dev->size;
+}
+
+static int _set_cca_threshold(ieee802154_dev_t *hal, int8_t threshold)
+{
+    (void)hal;
+    (void)threshold;
+
+    return 0;
```

Comment:
Since it's not implemented:

```suggestion
    return -ENOTSUP;
```